crates/hyperdrive-math/src/long/close.rs

```diff
@@ -115,4 +159,52 @@ mod tests {
         assert!(result.is_err());
         Ok(())
     }
+
+    // Tests market valuation against yield space valuation when closing a long
+    // with the minimum transaction amount.
+    #[tokio::test]
+    async fn test_calculate_value_long() -> Result<()> {
+        let tolerance = int256!(1e15);
+
+        // Fuzz the spot valuation and yield space valuation against each other.
+        let mut rng = thread_rng();
+        for _ in 0..*FAST_FUZZ_RUNS {
+            let state = rng.gen::<State>();
+            let bond_amount = state.minimum_transaction_amount();
+            let maturity_time = state.position_duration();
+            let current_time = rng.gen_range(fixed!(0)..=maturity_time);
+
+            // Ensure curve_fee is smaller than spot_price to avoid overflows
+            // on the yield space valuation
+            if state.curve_fee() > state.calculate_spot_price()? {
+                continue;
+            }
+
+            let yield_space_valuation = panic::catch_unwind(|| {
```

Comment:
there is not a benefit in catching panics and then immediately calling `unwrap`.

The function itself returns `Result<FixedPoint>`, and `panic::catch_unwind` catches any panics from inside the function and bundles them up into another Result, so that the new output type is `Result<Result<FixedPoitn>>`. This is commonly used to allow us to call the function and then handle a broken state gracefully. 

`unwrap` is meant to turn a `Result` into a `panic`. In other words, the outer unwrap is saying if there was a panic in `calculate_close_long` then panic now. The inner unwrap says if there was a `Result` in `calculate_close_long` then panic now. 

I think the idea here is the state is random so we might not be able to close a bond. If so, we want to skip this test. Assuming I'm right, here's what I would suggest we use a match statement to selectively skip this iteration of the test:

```rs
let yield_space_valuation = match panic::catch_unwind(|| { state.calculate_close_long(...) }) {
    Ok(valuation) => match valuation {
        Ok(valuation) => valuation, // no problems calling the fn
        Err(_) => continue, // fn threw an error
     },
    Err(_) => continue, // fn threw a panic
}
```

Reply:
I looked into this again (with a much better understanding thanks to your comments), and realized that most of what I had there wasn't necessary. We're trying to close the `minimum_transaction_amount` of bonds and, while the state is random, it's always a valid state, with enough breathing room that we would be able to close those bonds. 

There are some possible states that cause a `panic` in `calculate_close_long`, but lines 179-181 specifically skip those. Is that alright or would it be better practice to more generically catch panics and skip them here instead?